### DISCUSSION TITLE
How to find best abbreviation for sentence

### DISCUSSION THREAD

**Discussion by andr1972** [explosion/spaCy#12387] [Help: Best practices]
If I want compare two sentences I write
```
doc1 = nlp("Ala ma kota")
doc2 = nlp("Ala ma psa")
print(doc1.similarity(doc2))
```
but how do in this case:
There are about 10 sentences and 7-10 abbreviations of sentences, how to find best abbreviation for sentence?
This situation is in Wiktionary, for example word 'water'
1.A substance (of molecular formula H2O) found at room temperature and pressure as a clear liquid; it is present naturally as rain, and found in rivers, lakes and seas; its solid form is ice and its gaseous form is steam. 
2.The aforementioned liquid, considered one of the Classical elements or basic elements of alchemy. 
3.Water in a body; an area of open water.
......
Abbreviations:
liquid H₂O
mineral water 
spa water
one of the basic elements
body of water, or specific part of it

How to set threshold for warning if not found abbreviation for sentence, or two abbreviations are similar?

**Comment by thomashacker** [comment]
Hey, thanks for your question.

Can you further explain what you mean by finding the best _abbreviation for sentence_? Do you mean, a summary of a sentence?
Do you want to find the nearest sentence vector for a given sentence?

**Comment by andr1972** [comment]
For 'abbreviation' I take WIktionary abbreviation = phrase with mostly substring base text. 
First algorithm : longest common substring (is better than longest common subsequence for this problem)
